cognite_toolkit/_cdf_tk/commands/build.py

```diff
@@ -749,8 +749,15 @@ def local_variables(self) -> Mapping[str, str]:
     def update_local_variables(self, module_dir: Path) -> None:
         self._local_variables = _Helpers.create_local_config(self.variables_by_module_path, module_dir)
 
-    def create_file_name(self, filepath: Path) -> str:
-        return _Helpers.create_file_name(filepath, self.number_by_resource_type)
+    def create_file_name(self, filepath: Path, resource_directory: str) -> str:
+        filename = filepath.name
+        if filepath.suffix in EXCL_INDEX_SUFFIX:
+            return filename
+        # Get rid of the local index
+        filename = re.sub("^[0-9]+\\.", "", filename)
+        self.number_by_resource_type[resource_directory] += 1
+        filename = f"{self.number_by_resource_type[resource_directory]}.{filename}"
```

Comment:
This was the bug, we assumed that filepath.parent.name == resource_directory, which is not correct.